### PR TITLE
Feature: Better configuration

### LINKED ISSUE #32: New configuration system to allow for picking which metrics to upload
* Take arguments to the executable `--aggregate 5 --disable networkIO,diskIO`
Have an env variable to pass those options

Possible configuration:
- aggregation time: `--aggregate [minutes]` 0 for no aggregation. Default 1
- disable certain metrics `--disable [metricName],[metricName]`

### PR DESCRIPTION
**Pull request by timotheeguerin**
fix #32 

Provide ways to configure Batch Insights via command line arguments.
Is backward compatible with previous version which took positional arguments for pool id, node id and instrumentation key.
The deprecated args will be removed in version `2.0.0`

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+Available options
+    - diskIO
+    - diskUsage
+    - networkIO
+    - memory
+    - CPU
+    - GPU
+
+* `--aggregation <value>` Number in minutes to aggregate the data locally. Default to 1 minute
```
Comment (smith1511) [inline docs/configuration.md]: 'Defaults'

File: docs/configuration.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+Batch Insights provides various configuration option(Version `1.2.0` and above).
+
+
+#### `--poolID <value>` 
+Pool ID. Override pool ID provided by the `AZ_BATCH_POOL_ID` environment variable
+#### `--nodeID <value>` 
+Node ID. Override node ID provided by the `AZ_BATCH_NODE_ID` environment variable
+#### `--instKey <value>` 
+Instrumentation key. Application insights instrumentation key to emit the metrics
```
Comment (smith1511) [inline docs/configuration.md]: 'Insights' (capital I)

File: docs/configuration.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+#### `--disable <value>` 
+List of metrics comma separated to disable. e.g. `--disable networkIO,diskUsage`
```
Comment (smith1511) [inline docs/configuration.md]: Comma separated list of metrics to disable.

File: docs/configuration.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+Available options
```
Comment (smith1511) [inline docs/configuration.md]: Available metrics?

File: docs/configuration.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    - diskIO
+    - diskUsage
+    - networkIO
+    - memory
+    - CPU
+    - GPU
+
+* `--aggregation <value>` Number in minutes to aggregate the data locally. Default to 1 minute 
+
+Example: `--agregation 5` to aggreate for 5 minutes
```
Comment (smith1511) [inline docs/configuration.md]: typo: aggregate

File: docs/configuration.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+#### `--processes <value>` 
+List of process names to monitor comma separated.
```
Comment (smith1511) [inline docs/configuration.md]: Comma separated list of processes to monitor.

File: main.go [hunk 7]
```diff
@@ -1,48 +1,106 @@
...
+	})
+}
+
 func main() {
-	var appInsightsKey = os.Getenv("APP_INSIGHTS_INSTRUMENTATION_KEY")
-	var poolId = os.Getenv("AZ_BATCH_POOL_ID")
-	var nodeId = os.Getenv("AZ_BATCH_NODE_ID")
-	var processNamesStr = os.Getenv("AZ_BATCH_MONITOR_PROCESSES")
+	initLogger()
+	disableArg := flag.String("disable", "", "List of metrics to disable")
```
Comment (smith1511) [inline main.go]: Not sure if we own this, but it isn't really a flag, more an argument.

Comment (timotheeguerin) [reply main.go]: You mean the flag import?
That's the standard go argument parser

**Pull request merged**